src/beliefpropagation/beliefpropagation.jl

```diff
@@ -1,104 +1,82 @@
-function message_tensors(
-  tn::ITensorNetwork;
-  nvertices_per_partition=nothing,
-  npartitions=nothing,
-  subgraph_vertices=nothing,
-  kwargs...,
-)
-  return message_tensors(
-    partition(tn; nvertices_per_partition, npartitions, subgraph_vertices); kwargs...
-  )
+default_mt_constructor(inds_e) = ITensor[denseblocks(delta(inds_e))]
+default_bp_cache(ptn::PartitionedGraph) = Dict()
+function default_contractor(contract_list::Vector{ITensor})
+  return ITensor[normalize!(
+    contract(contract_list; sequence=contraction_sequence(contract_list; alg="optimal"))
+  )]
 end
 
-function message_tensors_skeleton(subgraphs::DataGraph)
-  mts = DataGraph{vertextype(subgraphs),vertex_data_type(subgraphs),ITensorNetwork}(
-    directed_graph(underlying_graph(subgraphs))
+function contract_to_MPS(contract_list::Vector{ITensor}; svd_kwargs...)
+  contract_kwargs = (;
+    alg="density_matrix",
+    output_structure=path_graph_structure,
+    contraction_sequence_alg="optimal",
+    svd_kwargs...,
   )
-  for v in vertices(mts)
-    mts[v] = subgraphs[v]
-  end
 
+  tn, _ = contract(ITensorNetwork(contract_list); contract_kwargs...)
+  mts = Vector{ITensor}(tn)
+  mts = normalize!.(mts)
```

Comment:
I don't think `Normalize(it::ITensor)` exists? Or at least it is not exported by `ITensors`

Reply:
Ok, we could do:
```julia
  mts .= normalize!.(copy.(mts))
```

Reply:
The main goal here is to not modify the tensor data in-place, and instead modify the collection storing the tensors.

I generally try to avoid in-place modification of tensor data unless it is really necessary for performance, since it makes code easier to reason about and can make the code less generic (for example if the tensor data can't be modified for some reason).